### PR TITLE
reposilite: init at 3.5.13

### PR DESCRIPTION
**Pull request by Jamalam360**
## Description of changes

Reposilite is a Maven server with a nice frontend. Homepage [here](https://reposilite.com).

They have an existing Nix guide [here](https://reposilite.com/guide/nixos), but I haven't been able to find any attempts at adding it to nixpkgs before. I will send a PR to update that page if this get merged. I also intend to write a NixOS service for Reposilite, but I was unsure if that should be included in this PR so I've held off.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

^ I was unsure if a test was required for a package this simple (a lot of the packages I looked at didn't seem to include one)

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Jamalam360** [comment]
Oops, fixed the format and squashed the format commit into the original maintainer commit.

File: pkgs/by-name/re/reposilite/package.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  installPhase = ''
+    mkdir -p $out/lib
+    cp $src $out/lib/reposilite
+
+    makeWrapper ${jre_headless}/bin/java $out/bin/reposilite \
+      --add-flags "-Xmx40m -jar $out/lib/reposilite"
+  '';
+
+  meta = with lib; {
+    description = "Lightweight and easy-to-use repository management software dedicated for the Maven based artifacts in the JVM ecosystem 📦";
```
Comment (JohnRTitor) [inline pkgs/by-name/re/reposilite/package.nix:25]: Let's not use emojis here

File: pkgs/by-name/re/reposilite/package.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{ stdenv, lib, fetchurl, makeWrapper, jre_headless }:
+
+stdenv.mkDerivation rec {
```
Comment (JohnRTitor) [inline pkgs/by-name/re/reposilite/package.nix:3]: Could you use `finalAttrs`?

**Comment by Jamalam360** [comment]
Thanks for the review @JohnRTitor, I think I have done `finalAttrs` correctly (based on how other nixpkgs packages have done it).

File: pkgs/by-name/re/reposilite/package.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  src = fetchurl {
+    url = "https://maven.reposilite.com/releases/com/reposilite/reposilite/${finalAttrs.version}/reposilite-${finalAttrs.version}-all.jar";
+    hash = "sha256-G9J1nOK1dM8XC+3Mj8uUnNvM5BiUPzOIRT/T5CZzvqo=";
+  };
+
+  dontUnpack = true;
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase = ''
```
Comment (Aleksanaa) [inline pkgs/by-name/re/reposilite/package.nix:16]:
```
runHook preInstall

... your code

runHook postInstall
```

**Comment by Jamalam360** [comment]
resolved, thanks :)

**Comment by JohnRTitor** [comment]
Result of `nixpkgs-review pr 320832` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>reposilite</li>
  </ul>
</details>

**Comment by NotAShelf** [comment]
Result of `nixpkgs-review pr 320832` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>reposilite</li>
  </ul>
</details>

**Review by NotAShelf — APPROVED** [review]

**Review by JohnRTitor — APPROVED** [review]
LGTM

**Pull request merged**